src/System.Reflection.Metadata.ILDasm/src/Decoder/ILDecoder.cs

```diff
@@ -0,0 +1,582 @@
+﻿using ILDasmLibrary.Instructions;
+using Microsoft.CSharp;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Decoding;
+using System.Reflection.Metadata.Ecma335;
+using System.Text;
+
+namespace ILDasmLibrary.Decoder
```

Comment:
Shouldn't the namespace match the assembly, e.g. `System.Reflection.Metatdata.ILDasm.Decoder`? This applies throughout. (As an aside, I left an overall comment about the use of "ILDasm").